app/models/ems_refresh/save_inventory.rb

```diff
@@ -101,7 +102,11 @@ def save_vms_inventory(ems, hashes, target = nil)
         found.raw_power_state = key_backup[:raw_power_state]
 
         link_habtm(found, key_backup[:storages], :storages, Storage)
-        link_habtm(found, key_backup[:security_groups], :security_groups, SecurityGroup)
+        # TODO(lsmola) NetworkManager, once all providers are converted, security groups relation of Vm will go away,
+        # being moved to NetworkPort, we will ignore here through associations
+        if found.try(:security_groups).try(:proxy_association).try(:reflection).try(:options).try(:[], :through) != :network_ports
```

Comment:
Yeah: don't do that. The very idea of this conditional is 100/10 horribles.

But if we're going to do it:
```ruby
security_groups_association = found.reflect_on_association(:security_groups)
if security_groups_association && security_groups_association.options[:through] # != .. if you feel it necessary
  # ..
end
```